raiden/lightclient/handlers/light_client_message_handler.py

```diff
@@ -299,7 +305,10 @@ def store_update_non_closing_balance_proof(cls, non_closing_balance_proof_data:
     def get_latest_light_client_non_closing_balance_proof(cls, channel_id: int, storage: SerializedSQLiteStorage):
         latest_update_balance_proof_data = storage.get_latest_light_client_non_closing_balance_proof(channel_id)
         if latest_update_balance_proof_data:
-            balance_proof = Unlock.from_dict(json.loads(latest_update_balance_proof_data[7]))
+            balance_proof_dict = json.loads(latest_update_balance_proof_data[7])
+            balance_proof = Unlock.from_dict(balance_proof_dict) \
+                if balance_proof_dict["type"] == "Secret" \
```

Comment:
Be careful here because the balance_proof_dict could have other case where is not Unlock or LockedTransfer, it could be LockExpired if i'm not wrong, if that's the case this call ```LockedTransfer.from_dict(balance_proof_dict)``` on the else statement will fail

Reply:
We were only supporting NonClosingBP for Unlock message (message_type = 'Secret') in LC.

I added support for LockedTransfer, but LockExpired (and there is also another one i don't remeber i think) are not possible messages for LC, at least not to be used as NonClosingBP. 

So this should not fail